### ISSUE TITLE
Isar doesn't retrieve records with where clause on composite keys

### ISSUE THREAD

**Issue by srix55** [isar/isar#383]
**Steps to Reproduce**
* Include 3 composite keys
* Retrieval using where on single or two composite keys results in empty lists

**Code sample**
Complete reproducible demo is at https://github.com/srix55/isar_composite_where_issue

```dart
// PlayerCollection
import 'package:isar/isar.dart';

part 'player_collection.g.dart';

@Collection()
@Name("Player")
class PlayerCollection {
  @Id()
  @Name("isarId")
  int? isarId;

  @Name("playerName")
  String playerName;

  @Name("playerId")
  @Index(type: IndexType.hash, unique: true, composite: [CompositeIndex('city'), CompositeIndex('playerName')])
  String playerId;

  @Name("city")
  String city;

  PlayerCollection(
      {required this.isarId,
      required this.playerId,
      required this.playerName,
      required this.city});
}
```
The test is at https://github.com/srix55/isar_composite_where_issue/blob/main/test/player_collection_test.dart
```dart
// Test
late Isar isar;

setUpAll(() {
  isar = SignageDb.getIsar();
  var p1 = PlayerCollection(isarId: 990, playerId: '200', playerName: 'Manuel Neuer', city: 'Munich');
  var p2 = PlayerCollection(isarId: 991, playerId: '200', playerName: 'Bernd Leno', city: 'Munich');
  var p3 = PlayerCollection(isarId: 992, playerId: '202', playerName: 'Kevin Trapp', city: 'Frankfurt');
  isar.writeTxnSync((isar) => isar.playerCollections.putAllSync([p1, p2, p3], replaceOnConflict: true));
});

test('getting records with filter', () { // PASSES
  var storedListWithFilter = isar.playerCollections.filter().playerIdEqualTo('200').and().cityEqualTo('Munich').findAllSync();
  expect(storedListWithFilter.isNotEmpty, true);
  expect(storedListWithFilter.length, 2);
});

test('getting records with single composite-index property', () { // FAILS
  var storedListWithSingleWhere = isar.playerCollections.where().playerIdEqualTo('200').findAllSync();
  expect(storedListWithSingleWhere.isNotEmpty, true);
  expect(storedListWithSingleWhere.length, 2);
});

test('getting records with two composite-indexes', () { // FAILS
  var storedListWithWhere = isar.playerCollections.where().playerIdCityEqualTo('200', 'Munich').findAllSync();
  expect(storedListWithWhere.isNotEmpty, true);
  expect(storedListWithWhere.length, 2);
});

test('getting records with three composite-indexes', () { // PASSES
  var storedListWithWhere = isar.playerCollections.where().playerIdCityPlayerNameEqualTo('200', 'Munich', 'Manuel Neuer').findAllSync();
  expect(storedListWithWhere.isNotEmpty, true);
  expect(storedListWithWhere.length, 1);
});
```

**Version**
 - Platform: Linux
 - Flutter version: 3.0.0
 - Isar version: 2.5.0
 - Isar Generator version: 2.5.0

**Comment by leisim** [comment]
I believe this is a duplicate of #372. Will be fixed in the next version

**Comment by leisim** [comment]
Fixed in 3.0.0-dev.2
